x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/planner/QueryFolderTests.java

```diff
@@ -65,6 +65,49 @@ public void testFoldingToLocalExecWithProject() {
         assertThat(ee.output().get(0).toString(), startsWith("keyword{f}#"));
     }
 
+    public void testFoldingToLocalExecBooleanAndNull_WhereClause() {
+        PhysicalPlan p = plan("SELECT keyword FROM test WHERE int > 10 AND null");
```

Comment:
Looks to me like this should be handled by the optimizer - namely it should evaluate the `AND` to false.

Reply:
Darn it, lost my comment - the issue is with `And` nullable which should return true if either one of its params is nullable not if both are. Once fixed, the nullable rule should kick in, nullify the end which would become a false query which would then be equivalent to false.

Reply:
If we do that then the `<expr> OR null` is not foldable so the `null` reached the QueryTranslator and cannot be translated. The problem is also that with `OR` we need to partially fold only the null side and eliminate the OR, by returning the non nullable side. It's related to: #35232